libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/logger/FileLogger.java

```diff
@@ -0,0 +1,257 @@
+     */
+    public void flushLog() throws IOException {
+        try {
+            file.clear();
+        } catch (IOException e) {
+            Log.e(TAG, "Failed to flush log file", e);
+        }
+    }
+
+    /**
+     * Returns the number of log lines stored in this file.
+     *
+     * @return Number of stored log lines.
+     */
+    public int getSize() {
+        return file.size();
+    }
+
+    /**
+     * Returns the maximum number of log lines that can be stored in this file.
+     *
+     * @return Maximum number of log lines.
+     */
+    public int getMaxSize() {
+        return maxSize;
+    }
+
+    /**
+     * Sets the maximum number of log lines that can be stored in this file.
+     *
+     * @param size Maximum number of log lines.
+     */
+    public synchronized void setMaxSize(int size) {
+        if (size < 0) {
+            size = 0;
+        }
+        maxSize = size;
+    }
+
+    /**
+     * Writes a log line to the file.
+     *
+     * @param logLine Log line.
+     */
+    public void addLogLine(String logLine) {
+        if (TextUtils.isEmpty(logLine)) {
+            return;
+        }
+        try {
+            while (getSize() >= maxSize) {
+                file.remove();
+            }
+            if (maxSize > 0) {
+                file.add(logLine.getBytes(UTF8));
+            }
+        } catch (Exception e) {
+            Log.e(TAG, "Failed to write log line", e);
+        }
+    }
+
+    /**
+     * Writes a list of log lines to the file.
+     *
+     * @param logLines Log lines.
+     */
+    public void addLogLines(List<String> logLines) {
+        if (logLines == null || logLines.size() == 0) {
+            return;
+        }
+        String[] logLinesArray = new String[logLines.size()];
+        logLines.toArray(logLinesArray);
+        addLogLines(logLinesArray);
+    }
+
+    /**
+     * Writes an array of log lines to the file.
+     *
+     * @param logLines Log lines.
+     */
+    public void addLogLines(String[] logLines) {
+        if (logLines == null || logLines.length == 0) {
+            return;
+        }
+        for (final String logLine : logLines) {
+            addLogLine(logLine);
+        }
+    }
+
+    /**
+     * Returns a single log line in FIFO.
+     *
+     * @return Log line.
+     */
+    public String readLogLine() {
+        String logLine = null;
+        try {
+            byte[] logLineBytes = file.peek();
+            if (logLineBytes != null && logLineBytes.length > 0) {
+                logLine = new String(logLineBytes, ASCII);
+            }
+        } catch (IOException e) {
+            Log.e(TAG, "Failed to read log line", e);
+        }
+        return logLine;
+    }
+
+    /**
+     * Returns a list of log lines in FIFO and removes them as they are being read.
+     *
+     * @param numLines Number of lines to be read.
+     * @return List of log lines.
+     */
+    public List<String> readAndRemoveLogLinesAsList(int numLines) {
```

Comment:
The idea is to provide a way to retrieve log lines for apps that want to upload them.